lib/ansible/modules/remote_management/cpm/cpm_status.py

```diff
@@ -0,0 +1,171 @@
+
+# Get firmware version
+- name: Get the firmware version of a given WTI device
+  cpm_status:
+    cpm_action: "firmware"
+    cpm_url: "192.168.0.158"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+
+# Get status output
+- name: Get the status output from a given WTI device
+  cpm_status:
+    cpm_action: "status"
+    cpm_url: "rest.wti.com"
+    cpm_username: "rest"
+    cpm_password: "restfulpassword"
+    use_https: true
+    validate_certs: true
+
+# Get Alarm output
+- name: Get the alarms status of a given WTI device
+  cpm_status:
+    cpm_action: "alarms"
+    cpm_url: "192.168.0.158"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: false
+"""
+
+RETURN = """
+data:
+    description: The output JSON returned from the commands sent
+    returned: always
+    type: str
+"""
+
+from ansible.module_utils.remote_management.cpm.cpm_common import request
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        cpm_action=dict(choices=['temperature', 'firmware', 'status', 'alarms'], required=True),
+        cpm_url=dict(type='str', required=True),
+        cpm_username=dict(type='str', required=True),
+        cpm_password=dict(type='str', required=True, no_log=True),
+        use_https=dict(type='bool', default=True),
+        validate_certs=dict(type='bool', default=True)
+    )
+
+    result = dict(
+        changed=False,
+        data=''
+    )
+
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    if module.check_mode:
+        return result
+
+    if module.params['use_https'] is True:
+        protocol = "https://"
+    else:
+        protocol = "http://"
+
+    if (module.params['cpm_action'] == 'temperature'):
+        result['data'] = request(module, protocol + module.params['cpm_url'] + "/api/v2/status/temperature", module.params['cpm_username'],
+                                 module.params['cpm_password'], 8)
+    elif (module.params['cpm_action'] == 'firmware'):
+        result['data'] = request(module, protocol + module.params['cpm_url'] + "/api/v2/status/firmware", module.params['cpm_username'],
+                                 module.params['cpm_password'], 8)
+    elif (module.params['cpm_action'] == 'status'):
+        result['data'] = request(module, protocol + module.params['cpm_url'] + "/api/v2/status/status", module.params['cpm_username'],
+                                 module.params['cpm_password'], 8)
+    elif (module.params['cpm_action'] == 'alarms'):
+        result['data'] = request(module, protocol + module.params['cpm_url'] + "/api/v2/status/alarms", module.params['cpm_username'],
+                                 module.params['cpm_password'], 8)
+    else:
+        module.fail_json(msg='Status command not recognized.', **result)
```

Comment:
This should not be necessary as validation is handled by the `choices` list in the `argument_spec`.